### ISSUE TITLE
Make filter applies for subfolders too

### ISSUE THREAD

**Issue by thumbapps** [tablacus/TablacusExplorer#77]
First arigato for the multi tab explorer :)

You frequently update TE but I don't see any improvement for the filter feature. Could you make it filter not just the current directory but also its subfolders?

I have to run File Explorer to do filter/search right now.

Regards,
Ronny
@thumbapps.org

**Comment by tablacus** [comment]
How about these add-ons?

1. Search Bar
2. Find files
3. Flat + Filter bar
4. Everything

Best regards,

**Comment by thumbapps** [comment]
Thanks. I'll try 'em and see which one can match FE's features.

Domo :)

**Comment by thumbapps** [comment]
I don't try Flat and Everything, but I decide to replace Filter bar with Search bar--it's faster than Find files.

Thanks 👍

**Comment by stefanocaselli** [comment]
Hello,
I want to use Search bar too, but I'm using multiple panels. Is it possibile to add a "Inner Search bar", just like the current "Inner Filter Bar" ?
Thanks.

**Comment by tablacus** [comment]
Hello,
I've created an add-on "Inner search bar".
Best regards,

**Comment by stefanocaselli** [comment]
Thank you!

2017-03-06 15:56 GMT+01:00 Gaku <notifications@github.com>:

**Comment by garry-ut99** [comment]
> tablacus: How about these add-ons?

> 1. Search Bar
- no automatic search support (like in Filter box or Everything standalone)
- no matched part highlighting support (link in Windows Explorer) or bolding (like in standalone Everything)
- no regex support
 
> 2. Find files
- obstructing and annoying popup appears
- no automatic search support (like in Filter box or Everything standalone)
- after pressing start - another obstructing and annoying progress popup appears
- no fast search support - slow searching - takes several seconds to finish searching
- no matched part highlighting support (like in Windows Explorer) or bolding (like in standalone Everything)
- no regex support

> 3. Flat + Filter bar

- when invocking Flat from tab context menu - it crashes:
"progress is empty or is not an object (00x800a138f) Line:33
![Image](https://github.com/user-attachments/assets/003990b1-9b48-452e-b8e3-100ee6968df6)
- (I guess) no matched part highlighting (like in Windows Explorer) or bolding (like in standalone Everything)

> 4. Everything integration
- no automatic search support (like in Filter box or Everything standalone)
- no matched part highlighting support (like in Windows Explorer) or bolding (like in standalone Everything)

As seen all of the above search/filter add-ons are dysfunctional and none of them can be called humanized and civilized. The only one application which seem do the job is standalone Everything launched separately, because it:
- supports automatic autosearch
- supports matched part highlighting (bold)
- supports subfolders searching
- supports regex
- supports fast searching (aka index)

But it lacks full integration with TE and need to be used separately which is cumbersome.

One idea to have a humanized and civilized search is to enhance further "Filter box" functionality.

It already has implemented:
- automatic search support
- regex support

There should be implemented further:
-  a subfolder searching - (for example a tickbox button available in Filter box which enables/disables subfolder searching on demand)
- matched part highlighting / bolding support

**Comment by ccfs** [comment]
I could not reproduce the crash of "Flat" add-on. However, please try this [temporary version](https://github.com/user-attachments/files/18900932/flat.zip) in which I have changed a doubtful part.

You can start a Windows Search by inputting a search criterion to Filter Bar and press Ctrl+Enter. Then you can refine the criterion, and the search will be automatic.

**Comment by garry-ut99** [comment]
> ccfs: However, please try this [temporary version](https://github.com/user-attachments/files/18900932/flat.zip) in which I have changed a doubtful part.

I've tried and it doesn't crash anymore, but on the other hand the add-on doesn't seem to work properly either: after invoking it from tab context menu, all files and folders in the tab are disappearing.

**Comment by ccfs** [comment]
Regarding Flat add-on problem:
* For diagnosis, please copy and paste your TE info from "Help > Tablacus Explorer" into your reply, together with screen captures (TE window and the tab context menu) showing how you get the empty tab.
* Do the folders and files in tab have the Hidden attribute?

**Comment by garry-ut99** [comment]
I don't think you need that information anymore, as I found out that a fresh install of latest TE + Flat works on Windows 10 22H2, but not in Windows 7 SP1, so this is something related to Win 7 SP1. 

Also on Windows 10, Flat doesn't crash even without your last patch of Flat, means the Flat crashing issue was Windows 7 related as well.

**Comment by ccfs** [comment]
@garry-ut99, I suggest to upgrade your IE to version 11 and retry. I am on Win7 SP1 + IE11. I don't have the above Flat add-on problem nor the Filter bar problem B.

**Comment by garry-ut99** [comment]
Yes, I've updated from IE8 to IE 11 and the issue is fixed.

**Comment by ccfs** [comment]
The [patch](https://github.com/tablacus/TablacusExplorer/issues/692) has been updated to better support Windows Search from Filter bar.

**Comment by garry-ut99** [comment]
I tested it and it works, but Windows Search is a joke: 
https://www.google.com/search?q=Windows+Search+is+a+joke.

For example having a file: `F12Tools.dll`, you type `Tools` and it finds nothing, until you add a goddamn `*`: `*Tools` , which in practice means you need to add two of them: `*Tools*` because you never know in the advance whether `F12` (or anything) will be at the beginning or at the end or at the both sides.

**Comment by ccfs** [comment]
On Windows 7, tick "Find partial matches" on "Control Panel > Folder Options > Search tab" to ensure partial name search. There is no such option on Windows 10, and name search is always partial.
![Image](https://github.com/user-attachments/assets/b506aeba-0b92-459a-8e5c-2e69c50eafdb)

You can change Alt+Up to something else you prefer using Mouse add-on.

**Comment by garry-ut99** [comment]
> ccfs: On Windows 7, tick "Find partial matches" (...)

I'm aware of it already,  as I do have it already ticked, since long time, and it doesn't help. 

However I visited options again to make sure, and nothing unusual - it was already ticked like I already said, but then after I closed options dialog by clicking "OK" it somehow now works, without even touching "Find partial matches" setting. So despite it was already ticked, it was broken, and simply visiting options and clicking OK fixed and reactivated it somehow. A strange glitch I suppose.

> ccfs: You can change Alt+Up to something else you prefer using Key add-on.

I tried "Right Alt+Up" as "something else" but it didn't work, I appended manually "Right" to "Alt+Up" so it became "Right Alt+Up", but it didn't work, regardless "Alt" is ticked as a modifier or not. It seems only "Alt" is supported as a modifer but not "Right Alt" separately, unless "Alt" modifer is supposed to work with both left and right Alt keys, then I don't know why it works only with left Alt. As far I see, other modifiers like "Ctrl + 1" work with both left and right Ctrl keys. I wonder whether is it a some bug or an incompatibility with my keyboard/language.
